### PR TITLE
Add client attribute 'floating_geometry'

### PR DESCRIPTION
**Pull request by t-wissmann**
Add a new attribute to access and write the floating geometry of a
client. In order to keep the diff small, I kept the name of the C++
variable in the source code, even though '...geometry' is more
appropriate.

In the diff, I replaced member write-access with a preparation of the
new Rectangle and a single write. This is inevitable anyway because the
operators * and -> of Attribute_<> only return const references (as they
should). Read-only member access is no problem; here, I only had to
replace '.' with '->'.

There is not much necessity to test the attribute itself, but instead,
it serves as a tool to test other features of hlwm (e.g. the pad and the
monitor geometry).

### CODE REVIEW

**Comment by t-wissmann** [comment]
I'm not sure what's the better name: `float_geometry` or `floating_geometry`? The latter is more consistent I think, but the former is shorter.

**Comment by t-wissmann** [comment]
> In the diff, I replaced member write-access with a preparation of the
> new Rectangle and a single write.

And as I expected, I confused `width` and `height` from time to time. One occurrence was just discovered by the test suite :-).. So I guess I should read the diff again later or tomorrow.

**Comment by t-wissmann** [comment]
> > In the diff, I replaced member write-access with a preparation of the
> > new Rectangle and a single write.
> 
> And as I expected, I confused `width` and `height` from time to time. One occurrence was just discovered by the test suite :-).. So I guess I should read the diff again later or tomorrow.

I've just checked the diff and didn't spot any further swaps of `width`/`height` or `x`/`y`

File: NEWS [hunk 1]
```diff
@@ -20,8 +20,9 @@ Current git version
     'class', 'geometry', 'winid')
   * The setting 'monitors_locked' is now explicitly an unsigned integer.
   * The setting 'default_frame_layout' now holds an algorithm name.
-  * New client attribute 'content_geometry' holding the geometry of the
-    application's content.
+  * New client attribute:
```
Comment (blarz) [inline NEWS:23]: Rather use the plural attributes here?

**Comment by blarz** [comment]
> I'm not sure what's the better name: `float_geometry` or `floating_geometry`? The latter is more consistent I think, but the former is shorter.

I think I like `floating_geometry` better, consistency is king ;)
Everything else in commands or attributes is referring to something *floating.

File: NEWS (hunk 1 above)
Comment (t-wissmann) [reply NEWS]: [x] fixed.

**Comment by t-wissmann** [comment]
> > I'm not sure what's the better name: `float_geometry` or `floating_geometry`? The latter is more consistent I think, but the former is shorter.
> 
> I think I like `floating_geometry` better, consistency is king ;)
> Everything else in commands or attributes is referring to something *floating.

I've renamed it to `floating_geometry` :)

File: NEWS (hunk 1 above)
Comment (blarz) [reply NEWS]: Thanks!

**Comment by blarz** [comment]
> > > I'm not sure what's the better name: `float_geometry` or `floating_geometry`? The latter is more consistent I think, but the former is shorter.
> > 
> > 
> > I think I like `floating_geometry` better, consistency is king ;)
> > Everything else in commands or attributes is referring to something *floating.
> 
> I've renamed it to `floating_geometry` :)

Thank you :)

**Review by blarz — APPROVED** [review]

**Pull request merged**